### PR TITLE
Add dotenv

### PR DESCRIPTION
**Pull request by lilyvc**
This Pull Request adds `dotenv-load` to the Mosaic site and makes the same changes to the site generator templates. 

Users can now set NODE_ENV and the mosaic fs will read these env vars.

### CODE REVIEW

File: packages/site/package.json [hunk 1]
```diff
@@ -23,9 +23,9 @@
     "e2e:codegen": "npx playwright codegen localhost:3000",
     "start": "next start",
     "gen:snapshot": "yarn mosaic build --out snapshots --name latest --config mosaic.config.js",
-    "serve:snapshot:file": "yarn cross-env MOSAIC_MODE='snapshot-file' concurrently --kill-others 'yarn dev'",
-    "serve:snapshot:s3": "yarn cross-env MOSAIC_MODE='snapshot-s3' concurrently --kill-others 'yarn dev'",
-    "serve": "concurrently --kill-others 'yarn dev' 'yarn mosaic serve -c ./mosaic.config.js' -p 8080"
+    "serve:snapshot:file": "yarn cross-env MOSAIC_MODE='snapshot-file' NODE_ENV='local' concurrently --kill-others 'yarn dev'",
```
Comment (joshwooding) [inline packages/site/package.json:26]: Do you still need these changes?

Comment (lilyvc) [reply packages/site/package.json:26]: yeah I think so otherwise NODE_ENV is undefined and it doesn't load env.local

Comment (joshwooding) [reply packages/site/package.json:26]: Have you tried without? dotenv-load should handle this

Comment (lilyvc) [reply packages/site/package.json:26]: ohhh it does that's cool, will remove

**Comment by DavieReid** [comment]
Not had a look at the code ye (sorry) but this was released yesterday [env.t3](https://env.t3.gg)

**Review by ZarkoPernar — APPROVED** [review]

**Comment by lilyvc** [comment]
> Not had a look at the code yet (sorry) but this was released yesterday [env.t3](https://env.t3.gg)
> 
> Looks perfect...

I think we could look into this in the future - for now would be good to get this in as we have had env vars in our mosaic config for a while which aren't being read

**Review by joshwooding — APPROVED** [review]

**Review by DavieReid — APPROVED** [review]

**Pull request merged**